lib/ActivityStreams.js

```diff
@@ -152,9 +152,11 @@ ActivityStreams.prototype = {
    * Process the passed in links, save them, get from cache and response to content.
    */
   _processAndSendLinks(links, responseType, append, worker, previewsOnly = false) {
+    const event = this._tabTracker.generateEvent({source: responseType});
     let processedLinks = this._previewProvider.processLinks(links);
-    this._previewProvider.asyncSaveLinks(processedLinks);
+    this._previewProvider.asyncSaveLinks(processedLinks, event);
```

Comment:
Regardless of the order, should be explicitly passing in an event object in the tests as well.  We don't want to wind up in a situation where the method definition expects two arguments and the tests only invoke one of them and are precariously dependent on that ordering to silently hide the second arg, and then someone comes along and changes something and can't figure out why things are failing.